app/org/maproulette/controllers/api/UserController.scala

```diff
@@ -60,34 +60,71 @@ class UserController @Inject()(userDAL: UserDAL,
-        Ok(Json.toJson(User.withDecryptedAPIKey(user)(crypto)))
-      } else if (user.isSuperUser) {
-        this.userDAL.retrieveByOSMID(userId, user) match {
-          case Some(u) => Ok(Json.toJson(User.withDecryptedAPIKey(u)(crypto)))
-          case None => this.userDAL.retrieveById(userId) match {
-            case Some(u) => Ok(Json.toJson(User.withDecryptedAPIKey(u)(crypto)))
-            case None => throw new NotFoundException(s"No user found with id '$userId'")
+    this.sessionManager.userAwareRequest { implicit user =>
+      var result : Option[Result] = None
+
+      user match {
+        case Some(requestUser) =>
+          // User can get itself
+          if (userId == requestUser.id || userId == requestUser.osmProfile.id) {
+            result = Option(Ok(Json.toJson(User.withDecryptedAPIKey(requestUser)(crypto))))
           }
-        }
-      } else {
-        throw new IllegalAccessException("Only super users have access to other user account information")
+          // Super User can get all users
+          else if (requestUser.isSuperUser) {
+            this.userDAL.retrieveByOSMID(userId, requestUser) match {
+              case Some(u) => result = Option(Ok(Json.toJson(User.withDecryptedAPIKey(u)(crypto))))
+              case None => this.userDAL.retrieveById(userId) match {
+                case Some(u) => result = Option(Ok(Json.toJson(User.withDecryptedAPIKey(u)(crypto))))
+                case None => throw new NotFoundException(s"No user found with id '$userId'")
+              }
+            }
+          }
+        case _ => None
+      }
+
+      result match {
+        case Some(result) => result
+        case _ =>
+          // User is not logged in or Not a super user or is Not requesting itself
+          // so we return a basic user object
+          val target = this.userDAL.retrieveByOSMID(userId, User.superUser) match {
+            case Some(u) => u
+            case None => this.userDAL.retrieveById(userId) match {
+              case Some(u) => u
+              case None => throw new NotFoundException(s"No user found with id '$userId'")
+            }
+          }
+
+          Ok(buildBasicUser(target))
       }
     }
   }
 
   def getUserByOSMUsername(username: String): Action[AnyContent] = Action.async { implicit request =>
-    this.sessionManager.authenticatedRequest { implicit user =>
-      if (user.name == username) {
-        Ok(Json.toJson(user))
-      } else {
-        // we don't need to check access here as the API only allows super users to make the call,
-        // so if not a super user, the correct IllegalAccessException will be thrown
-        this.userDAL.retrieveByOSMUsername(username, user) match {
-          case Some(u) => Ok(Json.toJson(u))
-          case None => throw new NotFoundException(s"No user found with OSM username '$username'")
-        }
+    this.sessionManager.userAwareRequest { implicit user =>
```

Comment:
Same comment here as the one above, I think we need to separate out retrieval of basic user information.